### ISSUE TITLE
HA cannot connect to Insteon Hub after some time

### ISSUE THREAD

**Issue by Goosemav12** [pyinsteon/pyinsteon#204]
Since May 7th when Insteon came back online, my connection between HA and Insteon has been very unreliable.  If I reboot the hub then it will work for some time but eventually loses connection and I start seeing the below errors in the logs.  I did see in another post mention Hub connectivity issues occur when two connections try to access the local HTTP server built into the Hub at the same time or if an existing connection does not close.  So HA is making a connection to the local HTTP on the hub and Insteon is as well and that's causing issues?  That doesn't make sense as I know HA supported the hub before insteon went down.

I currently cannot connect to the hub from a browser connection to ip:25105 which would make sense why the logs show cannot connect to host.  I have a dhcp reservation for the hub so the IP hasn't changed and the insteon app continues to work.

![image](https://user-images.githubusercontent.com/106084488/173588910-bd8703f7-8f6d-4ac9-9b7b-abe6d44e9e14.png)

**Comment by RPros** [comment]
I have been having this same issue since early mid May.  HomeAssistant looses connection to the hub and appears to lock port 25105 on the Insteon hub.  The odd item is rebooting the HomeAssistant host/vm resolves the issue (settings -> system-> Hardware -> three dots -> reboot host ) at least for a day or more.  Another tidbit of information, whenever the connection loss occurs the homeassistant container always has a connection to the hub in a FIN_WAIT1 state, never disappears and does not move to FIN_WAIT2 state. It's like that connection is never closing and somehow hanging tcp port 25105 on the insteon hub.  After the homeassistant OS reboot the insteon hub port 25105 is available again. 
**During issue:**
![image](https://user-images.githubusercontent.com/108558728/176981658-80837757-8ba0-405d-8239-d2677599e585.png)
**After Homeassistant OS reboot - working (Note the insteon hub was NOT rebooted)** note that "nc" can now establish a connection
![image](https://user-images.githubusercontent.com/108558728/176981672-ae74df77-7e9d-484b-b8f7-b1ff4811731f.png)

**Some other logs that happen when the issue is occurring:**
2022-06-30 22:40:17 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Client error: (<class 'asyncio.exceptions.TimeoutError'>)
2022-06-30 22:42:35 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 22:44:47 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 22:46:58 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 22:49:09 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 22:51:20 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 22:53:31 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Client error: (<class 'aiohttp.client_exceptions.ClientConnectorError'>) Cannot connect to host 10.0.0.135:25105 ssl:default [Connect call failed ('10.0.0.135', 25105)]
2022-06-30 22:55:50 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 22:58:01 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 23:00:12 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 23:02:23 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 23:04:34 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Hub write failure (ClientError)
2022-06-30 23:06:45 ERROR (MainThread) [pyinsteon.protocol.http_reader_writer] Client error: (<class 'aiohttp.client_exceptions.ClientConnectorError'>) Cannot connect to host 10.0.0.135:25105 ssl:default [Connect call failed ('10.0.0.135', 25105)]

**Below is my current version info:**
Home Assistant Core 2022.6.7
Home Assistant Supervisor 2022.05.3
Home Assistant OS 8.2

**Comment by Goosemav12** [comment]
Yes it appears either HA Insteon or the Hub doesn't close the connection.  I saw a post on the FB insteon user group where someone mentions that hoobs has created a connection watcher as a workaround for the time being.

**Comment by teharris1** [comment]
@Goosemav12 can you post the link to the connection watcher? I am curious how that works.

The issue in this post is not a new issue but I have never been able to get to the bottom of the root cause. It is definitely a problem with either the Hub or pyinsteon not closing the connection and restarting appears to be the only option. Any help with debugging this would be appreciated. 

It does appear that restarting the hub or restarting pyinsteon (ie. HA) fixes the issue. I would be curious if others see the same thing or if some people need to restart the Hub and others need to restart pyinsteon.

**Comment by RPros** [comment]
For me, I have to restart the Home Assistant OS (ova/VM), restarting the Home Assistant Core does not seem to release the connection in the FIN_WAIT1 state.  I do not have to restart the Insteon hub though.  If you let me know how to restart pyinsteon I can try that and let you know; the issue happens to me almost daily.  I went down a rabbit hole trying to forcibly close the connection but without tcpkill or ss on the container or HA OS I wasn't able to kill the FIN_WAIT1 connection. I even tried changing the /proc/sys/net/ipv4/tcp_max_orphans to 0 from 8192 and it didn't get closed.
The info I was able to find on the connection watcher that @Goosemav12 mentioned is at https://plugins.hoobs.org/plugin/homebridge-platform-insteonlocal under the "Connection Watcher" section near the bottom.

**Comment by teharris1** [comment]
So the Connection Watcher is for a different solution and is not applicable to how `pyinsteon` has implemented the connection. That solution is a native homekit solution addon which is trying to use a persistent connection. I tried that approach and it failed miserably. I am now using a pseudo restful approach where each request is a separate connection. This has been more reliable but still as good as I want. The only good news is it appears the homekit solution has run into the same issue so it is likely not specific to `pyinseon`. That does not help us right now however.

**Comment by tanbronson** [comment]
For what it's worth,  Insteon App must be keeping a persistent connection, because those times HA couldn't make a connection, the Insteon app continued to work....

(& thanks everyone for helping to chase this down)

**Comment by larsvilhuber** [comment]
I can confirm that I experience the same issues, and solve it by restarting HA 1x day and on-demand. That does have some negative effects on Insteon devices (reverting to some previous state).

HA log says

```
  File "/usr/local/lib/python3.10/site-packages/pyinsteon/protocol/protocol.py", line 142, in connection_lost
    if exc and exc.exception():
AttributeError: 'SerialException' object has no attribute 'exception'
```

```
Home Assistant 2022.9.7
```
